third_party/move/move-compiler-v2/transactional-tests/tests/no-v1-comparison/closures/mut_ref_downgrade.move

```diff
@@ -0,0 +1,26 @@
+//# publish
+module 0xc0ffee::m {
+    struct NoCopy has drop;
+
+    public fun assn<T: drop>(ref: &mut T, x: T){
+        *ref = x;
+    }
+
+    public fun foo() {
+        let x = NoCopy;
+
+        let a: ||u64 has drop = ||{
+            let NoCopy = x;
+            1
+        };
+
+        let b: ||u64 has drop + copy = ||1;
+
+        assn<||u64 has drop>(&mut b, a);
+
+        b();
+        b();
```

Comment:
nit: why do we need two calls to `b()`?

Reply:
That is what showcases the bug (that you can make copies of `NoCopy`).